### PR TITLE
Use cmp to make it easy to see what is broken in kfservice_controller_test

### PR DESCRIPTION
**Pull request by rakelkar**
Got want is so much easier than trying to cmp to dumps of json... (this test appears to have race and fails on occasion for me... this is the first step towards figuring out what is going on)

This change does need "github.com/google/go-cmp/cmp/cmpopts", so can't be merged until that is in!

/assign @yuzisun

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/207)

### CODE REVIEW

**Comment by rakelkar** [comment]
https://github.com/kubeflow/kfserving/issues/209

**Comment by ellis-bigelow** [comment]
@rakelkar, looks fine to me? Can you post before and after? The issue isn't super clear

File: pkg/controller/kfservice/kfservice_controller_test.go [hunk 1]
```diff
@@ -433,12 +436,13 @@ func TestCanaryReconcile(t *testing.T) {
...
+	g.Eventually(func() string {
 		kfsvc := &servingv1alpha1.KFService{}
 		err := c.Get(context.TODO(), canaryServiceKey, kfsvc)
 		if err != nil {
-			return nil
+			return err.Error()
 		}
-		return &kfsvc.Status
-	}, timeout).Should(testutils.BeSematicEqual(&expectedKfsvcStatus))
+		diff := cmp.Diff(&expectedKfsvcStatus, &kfsvc.Status, cmpopts.IgnoreTypes(apis.VolatileTime{}))
```
Comment (ellis-bigelow) [inline pkg/controller/kfservice/kfservice_controller_test.go]: Can we just return this inline?

**Comment by rakelkar** [comment]
@ellis-bigelow  I've posted what it looks like right now in the issue.. after my fix it will look like this:
-before
+after

**Comment by rakelkar** [comment]
The test is failing because my PR doesnt add the required dependency (need to do that in a separate PR):
github.com/google/go-cmp/cmp/cmpopts

**Comment by ellis-bigelow** [comment]
Gotcha. Needs rebase too. I'm happy approve as soon as those two are complete.

**Comment by johnugeorge** [comment]
@rakelkar needs rebase to get the presubmit fix

**Comment by rakelkar** [comment]
@ellis-bigelow @yuzisun this is ready to merge!

**Comment by yuzisun** [comment]
merging !
/lgtm
/approve

**Comment by ellis-bigelow** [comment]
Traveling this weekend. Will catch up tomorrow.

**Pull request merged**